### PR TITLE
Update GetKubeconfigForContext API to accept Project ID instead of Project Name

### PR DESCRIPTION
**Pull request by prkalle**
### What this PR does / why we need it

This PR updates `GetKubeconfigForContext` API  logic to use/accept Project ID instead of Project Name

### Which issue(s) this PR fixes

Fixes #

### Release note

```release-note
Update GetKubeconfigForContext API to accept Project ID instead of Project Name
```

### CODE REVIEW

File: config/tanzu_context.go [hunk 1]
```diff
@@ -21,6 +21,7 @@ import (
 // keys to Context's AdditionalMetadata map
 const (
 	OrgIDKey            = "tanzuOrgID"
+	OrgNameKey          = "tanzuOrgName"
```
Comment (marckhouzam) [inline config/tanzu_context.go:24]: Why this change?

**Review by marckhouzam — COMMENTED** [review]
So there is not change in functionality, just a rename of a variable, right?
That means that plugins calling `ForProject()` can use either the project name or project id and it will work, correct?
I'm asking to confirm this change is completely backwards-compatible.

File: config/tanzu_context.go (hunk 1 above)
Comment (prkalle) [reply config/tanzu_context.go:24]: There is requirement to add Organization name as well in the tanzu context metadata. Since this is minor change I added it as part of this PR.

**Comment by prkalle** [comment]
> So there is not change in functionality, just a rename of a variable, right? That means that plugins calling `ForProject()` can use either the project name or project id and it will work, correct? I'm asking to confirm this change is completely backwards-compatible.

The kubeconfig generated should have projectID as part of the cluster.server URL for UCP beta3. This would work for beta2 if plugins passing `project name` instead of `project id`. However the backward compatibility is not guaranteed as these API changes are meant for beta releases.

**Review by marckhouzam — APPROVED** [review]
LGTM
Thanks @prkalle
